### DISCUSSION TITLE
Checking cache for key without loading the value

### DISCUSSION THREAD

**Discussion by seyfeb** [ZiggyCreatures/FusionCache#87] [Q&A] [answered]
Is there a way to check if a value for some key has already been added to the cache without actually loading the data? 

Think of a (large) blob stored in cache where you are only interested in the fact _if_ it's there but do not want to to actually load the large file/object.

Using the `Get` or `TryGet` methods return the `MaybeValue` with the data attached (if the key was there) which I wouldn't need.

**Comment by jodydonetti** [comment, marked as answer]
Hi @seyfeb , and thanks for considering FusionCache!

The underlying 2nd level works via the standard `IDistributedCache` interface: this allows FusionCache to automatically work with any of the available implementations of a distributed cache in .NET (see a list [here](https://github.com/ZiggyCreatures/FusionCache/blob/main/docs/CacheLevels.md)).

The downside is that the methods at FusionCache disposal are only the ones defined in the `IDistributedCache` interface (see [here](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.caching.distributed.idistributedcache?view=dotnet-plat-ext-6.0#methods)) which are:
- Get/GetAsync
- Refresh/RefreshAsync
- Remove/RemoveAsync
- Set/SetAsync

As you can see there is not an available method to just check for the existence of an entry without getting the entry itself.

If you have any idea on how to bypass this limit I'm open to suggestions!

**ps:** I'm playing with the idea of adding support for custom distributed cache implementations different than `IDistributedCache`, but as of now these are just experiments and it's too early to get an idea of the potential result.

**Reply by seyfeb** [reply]
I'm using the Sqlite cache as a distributed cache. You're right, there is no way to check the existence in an IDistributedCache (and I don't see a way around that). Any extension is dependent on the specific implementation of the used IDistributedCache, so there probably is no generic way to add this.

**Reply by jodydonetti** [reply]
Yep, as long as FusionCache will use `IDistributedCache` as the common abstraction, that would be it.
We'll see if in a future version I'll be able to take a different approach and pull that off 🤞

ps: if the above is good to you, can you mark it as the answer? Thanks

**Reply by seyfeb** [reply]
Yes.. but than again, you can't simply plug in existing cache implementations as easy, you need them to support your interface :-/
